packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -48,6 +48,22 @@ export function compareNftMetadata(newNftMetadata: NftMetadata, nft: Nft) {
   return differentValues > 0;
 }
 
+/**
+ * Checks whether the existing nft object has all the keys of the new incoming nft metadata object
+ * @param newNftMetadata - New nft metadata object
+ * @param nft - Existing nft object to compare with
+ * @returns Whether the existing nft object has all the new keys from the new Nft metadata object
+ */
+export function hasNewCollectionFields(
+  newNftMetadata: NftMetadata,
+  nft: Nft,
+): boolean {
+  const keysNewNftMetadata = Object.keys(newNftMetadata.collection || {});
+  const keysExistingNft = new Set(Object.keys(nft.collection || {}));
```

Comment:
Are we attempting to check whether the `collection` properties here are undefined/null? If so it might express intent better to use `??`:

``` suggestion
  const keysNewNftMetadata = Object.keys(newNftMetadata.collection ?? {});
  const keysExistingNft = new Set(Object.keys(nft.collection ?? {}));

Reply:
Right! Will update this 🙏